mesonbuild/backend/vs2010backend.py

```diff
@@ -870,6 +870,46 @@ def _prettyprint_vcxproj_xml(self, tree, ofname):
             of.write(doc.toprettyxml())
         replace_if_different(ofname, ofname_tmp)
 
+    def gen_vcxproj_filters(self, target, ofname):
+        # Generate pitchfork of filters based on directory structure.
+        root = ET.Element('Project', {'ToolsVersion': '4.0',
+                                      'xmlns': 'http://schemas.microsoft.com/developer/msbuild/2003'})
+        filter_folders = ET.SubElement(root, 'ItemGroup')
+        filter_items = ET.SubElement(root, 'ItemGroup')
+        mlog.debug(f'Generating vcxproj filters {target.name}.')
+        found_folders = []
+        for i in target.sources:
```

Comment:
This will only end up pointing at static (non-generated sources). I think that's fine, but maybe others don't?

Reply:
I looked into doing stuff for generated sources, but they seemed to be `str` and not `mesonlib.File`, so when I tried to append, stuff didn't really work out... :S

Perhaps I am missing something obvious here, but I also think that this is fine for now.

Reply:
I think for that you want something like:
```python
for t in target.generated:
    for i in t.get_outputs():
        if isinstance(g, GeneratedList):
             fname = os.path.join(self.get_target_private_dir(target), i)
        else:
             fname = os.path.join(g.get_subdir(), i)
```